### ISSUE TITLE
Add `iframe` support for `hint-mode`

### ISSUE THREAD

**Issue by Ambrevar** [atlas-engineer/nyxt#1526]
Hinting on the duckduckgo iframe at https://nyxt.atlas.engineer/start does not work.

**Comment by Ambrevar** [comment]
@aartaka You merged some work on iframes recently, any chance it fixes this?

**Comment by aartaka** [comment]
> @aartaka You merged some work on iframes recently, any chance it fixes this?

No, the iframe work I've done was purely request-related. For this issue, the fix should be something close to the #1828 fix (which I started but haven't yet finished implementing).

**Comment by Ambrevar** [comment]
Label as dom?

**Comment by aadcg** [comment]
I don't think this is important to solve by 3.0.

**Comment by Ambrevar** [comment]
Agreed.

**Comment by aartaka** [comment]
It is inoortant! As important as #1828! We need our keyboard interfaces to work as good as mouse for Nyxt to be good enough of keyboard-driven browser. For that, we should make hints work in iframes and Shadow DOM, because more and more websites use the latter, while lots already rely of the former ヽ(ｏ`皿′ｏ)ﾉ

**Comment by aadcg** [comment]
I agree it's important, but I can't commit to it when there are more pressing issues.

**Comment by Ambrevar** [comment]
Would it be API-breaking to fix it after the 3.0 release?

**Comment by aadcg** [comment]
I would say it is not.

**Comment by aartaka** [comment]
Closing this as not planned [after some analysis of IFrame behaviors](https://github.com/atlas-engineer/nyxt/pull/2821#issuecomment-1559503465).

**Comment by jmercouris** [comment]
Interesting, I wonder how that works...

**Comment by aadcg** [comment]
Having been bothered by Google login iframes lately, I had a look at the issue again.

Accessing iframes from the same origin is trivial, but useless in this case.

Accessing iframes from distinct origin that are controlled by the same entity (still irrelevant) is possible via [postMessage](https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage).

Extensions like Saka keys or Vimium are able to hint iframes because they're using WebExtension-specific APIs that grant access to all frames ([content scrips](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Content_scripts)).

In conclusion, it seems that the only way is to resort to WebExtension APIs.

**Comment by jmercouris** [comment]
Unless we can figure out some magic to do this in the future, it is unplanned for now.
